### PR TITLE
feat!: Add ability to "includeDisabled" to all calls that get users

### PR DESCRIPTION
**Pull request by kekey1**
The parameter will only take effect if the user is ADMIN or ONC. Otherwise only Active users will be returned and the parameter will be ignored.

[#OCD-4713]

### CODE REVIEW

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/CertificationBodyController.java [hunk 1]
```diff
@@ -236,10 +241,16 @@ public String deleteUserFromAcb(@PathVariable final Long acbId, @PathVariable fi
     })
     @RequestMapping(value = "/{acbId}/users", method = RequestMethod.GET,
             produces = "application/json; charset=utf-8")
-    public @ResponseBody UsersResponse getUsers(@PathVariable("acbId") final Long acbId)
+    public @ResponseBody UsersResponse getUsers(@PathVariable("acbId") Long acbId,
+            @Parameter(description = "Whether to include users whose accounts have been marked as disabled. The parameter only affects the response when called by an authenticated ADMIN or ONC user.",
+                allowEmptyValue = true, in = ParameterIn.QUERY, name = "includeDisabled")
+            @RequestParam(value = "includeDisabled", required = false, defaultValue = "false") String includeDisabled)
```
Comment (tmy1313) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/CertificationBodyController.java:253]: Shouldn't this be a `boolean`?

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/DeveloperController.java [hunk 2]
```diff
@@ -278,10 +283,14 @@ public PermissionDeletedResponse deleteUserFromDeveloper(@PathVariable Long deve
             })
     @RequestMapping(value = "/{developerId}/users", method = RequestMethod.GET,
             produces = "application/json; charset=utf-8")
-    public @ResponseBody UsersResponse getUsers(@PathVariable("developerId") Long developerId)
-            throws InvalidArgumentsException, EntityRetrievalException {
+    public @ResponseBody UsersResponse getUsers(@PathVariable("developerId") Long developerId,
+        @Parameter(description = "Whether to include users whose accounts have been marked as disabled. The parameter only affects the response when called by an authenticated ADMIN or ONC user.",
+            allowEmptyValue = true, in = ParameterIn.QUERY, name = "includeDisabled")
+        @RequestParam(value = "includeDisabled", required = false, defaultValue = "false") String includeDisabled)
```
Comment (tmy1313) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/DeveloperController.java:291]: Shouldn't this be a `boolean`?

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java [hunk 3]
```diff
@@ -478,12 +482,16 @@ public DeletedUser deleteUser(@PathVariable("userId") Long userId) throws UserRe
             })
     @RequestMapping(value = "", method = RequestMethod.GET, produces = "application/json; charset=utf-8")
     @PreAuthorize("isAuthenticated()")
-    public @ResponseBody UsersResponse getUsers() {
+    public @ResponseBody UsersResponse getUsers(
+            @Parameter(description = "Whether to include users whose accounts have been marked as disabled. The parameter only affects the response when called by an authenticated ADMIN or ONC user.",
+                allowEmptyValue = true, in = ParameterIn.QUERY, name = "includeDisabled")
+            @RequestParam(value = "includeDisabled", required = false, defaultValue = "false") String includeDisabled) {
```
Comment (tmy1313) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java]: Shouldn't this be a `boolean`?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/auth/UserManager.java [hunk 4]
```diff
@@ -164,14 +170,18 @@ public void delete(UserDTO user) throws UserRetrievalException, ActivityExceptio
...
-            + "T(gov.healthit.chpl.permissions.domains.SecuredUserDomainPermissions).GET_BY_PERMISSION)")
-    public List<UserDTO> getUsersWithPermission(String permissionName) {
-        return userDAO.getUsersWithPermission(permissionName);
+    public List<UserDTO> getAll(Boolean includeDisabled) {
+        if (!resourcePermissionsFactory.get().isUserRoleAdmin()
+                && !resourcePermissionsFactory.get().isUserRoleOnc()) {
+            includeDisabled = false;
+        }
+        List<UserDTO> allUsers = userDAO.findAll();
+        if (includeDisabled == null || BooleanUtils.isFalse(includeDisabled)) {
```
Comment (tmy1313) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/auth/UserManager.java:180]: Do we need to use `BooleanUtils`?  Also, in couple other places.

**Review by tmy1313 — CHANGES_REQUESTED** [review]

**Review by tmy1313 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by tmy1313 — APPROVED** [review]

**Pull request merged**